### DISCUSSION TITLE
wayvnc keeps closing (or crashing) on RPi OS x64

### DISCUSSION THREAD

**Discussion by m6tth3w** [any1/wayvnc#282] [Q&A] [answered]
Hey,

in advance, i would like to apologize if my question is not appropriate, but i am trying to find a solution for a couple of weeks now without success.

I'm using a fresh install of the 64 bit version of the Raspberry Pi OS on a RPi5 8 GB (tried to reinstall 2 times to see if i have done something wrong), but my wayvnc server keeps closing after some time. Sometimes it works for a day or so, sometimes for hours. I don't know the reason why it is closing.

I'm using RealVNC Viewer to connect to it, but i think that can't be the problem, because i have a RPi4 too, for another purpose, and that runs just well, can connect to vnc server without any problem, and it isn't keep closing on that system (also 64 bit and bookworm fresh install and updates).

If i re-enable the VNC service in raspi-config, it works again, or if i restart the system, but the problem remains, and it closes or crashes after a while.
I didn't do too much 'modification', installs on the system (except from the updates of course). I am using a minidlna service 7/24 on it on my local network, but that's all.

I also tried to look for a log for answers on the system for wayvnc but haven't found a file location for it. Even searched the internet where can be one found without success.
I'm a basic linux user, i know a few things on CLI so i'm not a complete beginner but also not a power user / advanced user.

So my question is what can be the problem, or how could i find out why it keeps stopping? And of course how to solve the problem without constantly re-enabling VNC service or restarting the system. I also found a way to start it through ssh i think, but that's not a solution either to restart always.

Thanks for any help in advance.

Greetings,
Matt

**Comment by any1** [comment]
See https://github.com/raspberrypi/bookworm-feedback/issues/42

**Comment by m6tth3w** [comment]
My RPi5 runs always with hdmi connected, though it is connected to a TV where i can switch outputs, and i do it when i want to watch TV or do other things and then the output changes (or might change). So i assume then it will change to headless mode (not sure but might be), and therefore wayvnc stops.

So i have created wayvnc.service and wayvnc-control.service in /usr/lib/systemd/system/ folder with the following content:

```
# wayvnc.service

[Unit]
Description=VNC Server
Documentation=man:wayvnc
After=network.target
ConditionPathExists=/etc/wayvnc/config

[Service]
ExecStart=/usr/bin/wayvnc -grD -C/etc/wayvnc/config
Type=simple
Restart=always

[Install]
WantedBy=multi-user.target
```

```
# wayvnc-control.service

[Unit]
Description=VNC Control Service
After=wayvnc.service
BindsTo=wayvnc.service
ConditionPathExists=/etc/wayvnc/config

[Service]
Type=simple
ExecStart=/home/m6tth3w/wayvnc/auto-attach.py # in wayvnc example scripts
Restart=always

[Install]
# Note: Needs to be enabled so that it's auto-run by starting wayvnc.
WantedBy=wayvnc.service
```

I had to do some changes:
I haven't got `wayvnc` folder in my home folder, so i changed that with the binary of wayvnc in `/usr/bin/` (I haven't built it myself, so i assume that's why it wasn't there).
Also haven't got `wayvnc` folder in `/etc/` so i created it and copied the contents of `~/.config/wayvnc/` . Hopefully those are the files that has to be there.
And i copied your example python file (`auto-attach.py`) to my home folder's newly created wayvnc folder (`~/wayvnc/auto-attach.py`), and chmod 777 it. (no idea if it is needed, though i thought it's some kind of executable, so why not. :D

Hopefully these are the right moves i had to do based on your description for detached mode (https://github.com/any1/wayvnc/pull/278).

Restarted my RPi5, now vnc server runs. Hopefully it stays that way if i made things right (the above mentioned changes i meant).

**Reply by any1** [reply, marked as answer]
Detached mode is a new feature that's only available on git master. I will be creating a debian package that sets this all up correctly, but haven't found the time to do so yet.

**Reply by m6tth3w** [reply]
Oh i see. Thanks for the information and sorry for the misunderstanding. It's ok, i'm not in a rush, just good to know this is an issue and will be sooner or later a solution for it.
I can wait.

Thanks again.

**Comment by m6tth3w** [comment]
Looks like something is not right. Just as i started my TV, wayvnc stopped. :S

What did i do wrong? If someone could help, would be really appreciated.

Also if i try to restart with `/usr/bin/wayvnc --render-cursor --keyboard=hu`, it says:
`ERROR: ../src/main.c: 402: WAYLAND_DISPLAY is not set in the environment`
`ERROR: ../src/main.c: 1495: Failed to initialise wayland`

Is there an easier way to restart it without setting those environment variables manually, or should i restart it using `raspi-config` and enable VNC service (or restart the OS)?

Thanks in advance.
